### PR TITLE
DataSources: Review indicator fix

### LINKED ISSUE #2059: Add message colors for comments and National data
@minotogna 
Currently, there is no symbol would appear if one makes comments in the **Comments** area or **National Data**.

### PR DESCRIPTION
**Pull request by sorja**
Resolves #2059 

Adds dynamic data sources in summary and status for showing review indicator correctly

https://user-images.githubusercontent.com/5508251/218269280-31e9da02-0e3f-4b0a-addc-28a037656e63.mov

### CODE REVIEW

File: src/client/pages/AssessmentSection/Descriptions/Description/DataSources/DataSourceRow.tsx [hunk 1]
```diff
@@ -48,7 +47,7 @@ const DataSourceRow: React.FC<Props> = (props: Props) => {
       ))}
 
       <DataColumn className="data-source-review-indicator">
-        {!disabled && <ReviewIndicator title={title} topicKey={`dataSource-${countryIso}-${sectionName}-${index}`} />}
+        {!isDataLocked && dataSource.uuid && <ReviewIndicator title={title} topicKey={dataSource.uuid} />}
```
Comment (sorja) [inline src/client/pages/AssessmentSection/Descriptions/Description/DataSources/DataSourceRow.tsx:50]: Hide review indicator from placeholder

File: src/client/store/ui/assessmentSection/actions/copyPreviousDatasources.ts [hunk 2]
```diff
@@ -26,7 +27,9 @@ export const copyPreviousDatasources = createAsyncThunk<
       updateDescription({
         value: {
           ...currentValue,
-          dataSources: value.dataSources?.map(({ fraVariables: _fraVariables, ...rest }: DataSource) => rest),
+          dataSources: value.dataSources?.map(({ fraVariables: _fraVariables, ...rest }: DataSource) => {
+            return { ...rest, uuid: UUIDs.v4() }
```
Comment (sorja) [inline src/client/store/ui/assessmentSection/actions/copyPreviousDatasources.ts:31]: On copy, update copied rows UUIDs

File: src/server/repository/assessmentCycle/messageTopicUser/getReviewStatus.ts [hunk 3]
```diff
@@ -17,42 +17,41 @@ export const getReviewStatus = async (
...
-        left join ${cycleSchema}.message_topic_user mtu
-          on mtu.topic_id = m.topic_id
-          and mtu.user_id = $2
-        left join ${cycleSchema}.message_topic mt
-          on mt.id = m.topic_id
-      where mt.country_iso = $3
+        with data_source_row_uuids as (select jsonb_array_elements(value -> 'dataSources') ->> 'uuid' as uuid
+                                       from ${cycleSchema}.descriptions d
+                                       where d.section_name = $1
+                                         and d.value -> 'dataSources' is not null),
```
Comment (sorja) [inline src/server/repository/assessmentCycle/messageTopicUser/getReviewStatus.ts:23]: These lines were included

File: src/server/repository/assessmentCycle/messageTopicUser/getReviewStatus.ts [hunk 4]
```diff
@@ -17,42 +17,41 @@ export const getReviewStatus = async (
...
+                   where not m.deleted
+                     and (mt.key in (select r.uuid::text
+                                     from ${schemaName}.row r
+                                              left join ${schemaName}."table" t
+                                                        on t.id = r.table_id
+                                              left join ${schemaName}.table_section ts
+                                                        on ts.id = t.table_section_id
+                                              left join ${schemaName}.section as s
+                                                        on s.id = ts.section_id
+                                     where s.props ->> 'name' = $1) or mt.key in (select uuid from data_source_row_uuids))
```
Comment (sorja) [inline src/server/repository/assessmentCycle/messageTopicUser/getReviewStatus.ts:39]: And the alias query was included here

File: src/server/repository/assessmentCycle/messageTopicUser/getReviewSummary.ts [hunk 5]
```diff
@@ -17,50 +17,57 @@ export const getReviewSummary = async (
...
+        with data_source_uuids as (select country_iso,
+                                          section_name,
+                                          jsonb_array_elements(value -> 'dataSources') ->> 'uuid' as uuid
+                                   from ${cycleSchema}.descriptions d
+                                   where d.value -> 'dataSources' is not null),
+             data_source_rows as (select dsr.uuid::uuid as row_uuid,
+                                         s.id           as sub_section_id,
+                                         s.parent_id
+                                  from data_source_uuids dsr
+                                           left join ${schemaName}.section s on s.props ->> 'name' = dsr.section_name),
```
Comment (sorja) [inline src/server/repository/assessmentCycle/messageTopicUser/getReviewSummary.ts:29]: These lines were included

File: src/server/repository/assessmentCycle/messageTopicUser/getReviewSummary.ts [hunk 6]
```diff
@@ -17,50 +17,57 @@ export const getReviewSummary = async (
...
+                            left join ${schemaName}."table" t
+                                      on t.id = r.table_id
+                            left join ${schemaName}.table_section ts
+                                      on ts.id = t.table_section_id
+                            left join ${schemaName}.section s
+                                      on s.id = ts.section_id
+                   where r.props -> 'cycles' ? '5883400e-988d-45e5-9b65-fff6715c25b7'
+                   union all
+                   select *
+                   from data_source_rows),
```
Comment (sorja) [inline src/server/repository/assessmentCycle/messageTopicUser/getReviewSummary.ts:41]: And the alias query was included here

**Review by minotogna — APPROVED** [review]

**Pull request merged**